js/aax.js

```diff
@@ -1902,6 +1903,94 @@ module.exports = class aax extends Exchange {
         return this.parseDepositAddress (data, currency);
     }
 
+    async fetchDeposits (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            // status Not required -  Deposit status, "1: pending,2: confirmed, 3:failed"
+            // currency: Not required -  String Currency
+            // startTime Not required Integer Default: 90 days from current timestamp.
+            // endTime Not required Integer Default: present timestamp.
+        };
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['currency'] = currency['id'];
+        }
+        if (since !== undefined) {
+            request['startTime'] = since; // default 90 days
```

Comment:
Their docs say, that startTime and endTime are in seconds. While in CCXT the `since` argument is in milliseconds, so we have to divide by a thousand here and round to the nearest integer:

```JavaScript
        if (since !== undefined) {
            request['startTime'] = parseInt (since / 1000); // default 90 days
        }
```

Also, apart from the above, they say:

> _Please notice the default `startTime` and `endTime` to make sure that time interval is within 0-90 days.If both `startTime` and `endTime` are sent, time between `startTime` and `endTime` must be less than 90 days._

What happens if the user sends a `since` value that is further than 90 days into the past? If they return a proper response, then we should be ok. However, in some cases we have to adjust the endTime as well, if they don't calculate the time range automatically:

```JavaScript
if (since !== undefined) {
    const startTime = parseInt (since / 1000);
    request['startTime'] = startTime;
    request['endTime'] = this.sum (startTime, 90 * 24 * 60 * 60)
}
```

↑ That requires running some testing requests, that can be done with the cli.js example. Please, let me know if you can resolve this aspect. Thanks again for your involvement!

Reply:
Great catch!
I haven't used cli.js example yet. I'll try it and get back to you on the testing requests

Reply:
I performed the following tests, and all performed as expected:
- `since` timestamp older than 90 days, returned a valid response, filtering out more recent transactions
- `since` timestamp from 5 days ago, returned a valid response
- `since` timestamp from a couple seconds ago, returned a valid response filtering out older transactions.